cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -23,6 +23,12 @@ include(CheckCXXSourceCompiles)
 
 message(STATUS "System processor: ${CMAKE_SYSTEM_PROCESSOR}")
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
```

Comment:
Do we need this check?
In general, we can't mix C/C++ compilers from different vendores, right?
(Can we mix g++ and clang?)

Reply:
Yea mixing g++ and clang is the problem. If you just set `-DCMAKE_CXX_COMPILER=clang++` then clang++ will be used for cpp files and gcc will be used for c (assuming gcc is the default)

Reply:
but wouldn't the following catch that?
`if(NOT CMAKE_CXX_COMPILER_ID STREQUAL CMAKE_C_COMPILER_ID)`
Why only perform the check if the `CMAKE_CXX_COMPILER_ID` is clang?

Reply:
Ah I see what you are saying. I don't know CMake well enough to know if those labels are always expected to be the same. Happy to make that change.

This originally started by locating the "bug" in SetupCxxFlags.cmake, hence the condition in the first place